### PR TITLE
Classifying Digits using scikit-learn

### PR DESCRIPTION
**Pull request by nitbix**

### CODE REVIEW

**Comment by SeanTAllen** [comment]
This needs a README or something for how to set up and run. What its for. The blog post its related to etc.

File: sklearn-example/digits.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+@wallaroo.decoder(header_length=4, length_fmt=">I")
+def decode(bs):
+    return pickle.loads(bs)
+
+
+@wallaroo.computation(name="PCA")
+def pca_transform(x):
+    return pca.transform([x])
+
```
Comment (SeanTAllen) [inline sklearn-example/digits.py]: should be newline here

File: sklearn-example/digits.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@wallaroo.computation(name="Logistic Regression")
+def logistic_classify(x):
+    return logistic.predict(x)
+
+@wallaroo.encoder
```
Comment (SeanTAllen) [inline sklearn-example/digits.py]: should be newline here

File: sklearn-example/sender.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import struct
+from sklearn import linear_model, decomposition, datasets
+
+import cPickle as pickle
+
+def send_message(conn, x):
+    msg = pickle.dumps(x)
+    conn.sendall(struct.pack('>I', len(msg)))
+    conn.sendall(msg)
+
```
Comment (SeanTAllen) [inline sklearn-example/sender.py]: should be newline here

File: sklearn-example/sender.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import sys
+import socket
+import struct
+from sklearn import linear_model, decomposition, datasets
+
+import cPickle as pickle
```
Comment (SeanTAllen) [inline sklearn-example/sender.py]: any reason why this import is off on its own?

File: sklearn-example/digits.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import struct
+from enum import IntEnum
```
Comment (aturley) [inline sklearn-example/digits.py]: Do we need to import `IntEnum`, `re`, and `json`? It doesn't look like they're used anywhere.

File: sklearn-example/train_models.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/python
+
+import numpy as np
```
Comment (aturley) [inline sklearn-example/train_models.py]: Is `numpy` used anywhere?

File: sklearn-example/README.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+# Log Files
```
Comment (SeanTAllen) [inline sklearn-example/README.md]: why is this "Log Files"

File: sklearn-example/README.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+This is an example of an application that classifies streams of hand-written digits. This example is discussed in the [A scikit-learn pipeline in Wallaroo](http://blog.wallarolabs.com/) blog post. For details on how the application works, please refer to the blog post.
```
Comment (SeanTAllen) [inline sklearn-example/README.md]: blog post link should be:

https:/blog.wallaroolabs.com/2018/02/a-scikit-learn-pipeline-in-wallaroo/

File: sklearn-example/README.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
+# Classifyin Digits using scikit-learn
+
+This is an example of an application that classifies streams of hand-written digits. This example is discussed in the [A scikit-learn pipeline in Wallaroo](http://blog.wallarolabs.com/) blog post. For details on how the application works, please refer to the blog post.
```
Comment (SeanTAllen) [inline sklearn-example/README.md]: https://blog.wallaroolabs.com/2018/02/a-scikit-learn-pipeline-in-wallaroo/

File: sklearn-example/README.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
+# Classifying Digits using scikit-learn
+
+This is an example of an application that classifies streams of hand-written digits. This example is discussed in the [A scikit-learn pipeline in Wallaroo](http://blog.wallarolabs.com/) blog post. For details on how the application works, please refer to the blog post.
+
+You will need a working [Wallaroo Python API](/book/python/intro.md).
```
Comment (SeanTAllen) [inline sklearn-example/README.md]: This should be:

https://docs.wallaroolabs.com/book/python/intro.html

File: sklearn-example/README.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
+# Classifying Digits using scikit-learn
+
+This is an example of an application that classifies streams of hand-written digits. This example is discussed in the [A scikit-learn pipeline in Wallaroo](http://blog.wallarolabs.com/) blog post. For details on how the application works, please refer to the blog post.
```
Comment (SeanTAllen) [inline sklearn-example/README.md]: https://blog.wallaroolabs.com/2018/02/a-scikit-learn-pipeline-in-wallaroo/

**Pull request merged**